ocpmodels/datasets/oc22_lmdb_dataset.py

```diff
@@ -0,0 +1,244 @@
+                self.envs[db_idx]
+                .begin()
+                .get(f"{self._keys[db_idx][el_idx]}".encode("ascii"))
+            )
+            data_object = pyg2_data_transform(pickle.loads(datapoint_pickled))
+            data_object.id = f"{db_idx}_{el_idx}"
+        else:
+            datapoint_pickled = self.env.begin().get(self._keys[idx])
+            data_object = pyg2_data_transform(pickle.loads(datapoint_pickled))
+
+        if self.transform is not None:
+            data_object = self.transform(data_object)
+        # make types consistent
+        sid = data_object.sid
+        if isinstance(sid, torch.Tensor):
+            sid = sid.item()
+            data_object.sid = sid
+        if "fid" in data_object:
+            fid = data_object.fid
+            if isinstance(fid, torch.Tensor):
+                fid = fid.item()
+                data_object.fid = fid
+
+        if hasattr(data_object, "y_relaxed"):
+            attr = "y_relaxed"
+        elif hasattr(data_object, "y"):
+            attr = "y"
+        # if targets are not available, test data is being used
+        else:
+            return data_object
+
+        # convert s2ef energies to raw energies
+        if attr == "y":
+            # OC20 data
+            if "oc22" not in data_object and self.train_total:
+                randomid = f"random{sid}"
+                data_object[attr] += self.oc20_ref[randomid]
+                data_object.nads = 1
+                data_object.oc22 = 0
+
+        # convert is2re energies to raw energies
+        else:
+            if "oc22" not in data_object and self.train_total:
+                randomid = f"random{sid}"
+                data_object[attr] += self.oc20_ref[randomid]
+                del data_object.force
+                del data_object.y_init
+                data_object.nads = 1
+                data_object.oc22 = 0
+
+        if self.lin_ref is not False:
+            lin_energy = sum(self.lin_ref[data_object.atomic_numbers.long()])
+            data_object[attr] -= lin_energy
+
+        # to jointly train on oc22+oc20, need to delete these oc20-only attributes
+        # ensure otf_graph=1 in your model configuration
+        if "edge_index" in data_object:
+            del data_object.edge_index
+        if "cell_offsets" in data_object:
+            del data_object.cell_offsets
+        if "distances" in data_object:
+            del data_object.distances
+
+        return data_object
+
+    def connect_db(self, lmdb_path=None):
+        env = lmdb.open(
+            str(lmdb_path),
+            subdir=False,
+            readonly=True,
+            lock=False,
+            readahead=False,
+            meminit=False,
+            max_readers=1,
+        )
+        return env
+
+    def close_db(self):
+        if not self.path.is_file():
+            for env in self.envs:
+                env.close()
+        else:
+            self.env.close()
+
+
+class SinglePointLmdbDataset(LmdbDataset):
```

Comment:
Do we need `SinglePointLmdbDataset` and `TrajectoryLmdbDataset` here? The definitions seem identical to the ones in [lmdb_dataset.py](https://github.com/Open-Catalyst-Project/ocp/blob/main/ocpmodels/datasets/lmdb_dataset.py#L126-L143).